test/parallel/test-crypto-authenticated.js

```diff
@@ -47,7 +47,39 @@ var TEST_CASES = [
     iv: '0e1791e9db3bd21a9122c416', plain: 'Hello node.js world!',
     password: 'very bad password', aad: '63616c76696e',
     ct: 'DDA53A4059AA17B88756984995F7BBA3C636CC44',
-    tag: 'D2A35E5C611E5E3D2258360241C5B045', tampered: false }
+    tag: 'D2A35E5C611E5E3D2258360241C5B045', tampered: false },
+  // 16 byte IV
+  { algo: 'aes-128-gcm',
+    key: '6970787039613669314d623455536234',
+    iv: 'cda28f25c83c80c5505692a4920e8ae7', plain: 'Hello World!',
+    ct: 'FCDC265C8B9109E24746C2EF',
+    tag: 'D0E578906E946B30E347EB44B42686C2', tampered: false },
+  { algo: 'aes-128-gcm',
+    key: '6970787039613669314d623455536234',
+    iv: 'cda28f25c83c80c5505692a4920e8ae7', plain: 'Hello World!',
+    ct: 'FCDC265C8B9109E24746C2EF', aad: 'ace0dd5ad2bf',
+    tag: 'D00353D63C3A170F0B682E2EDFF5908D', tampered: false },
+  { algo: 'aes-256-gcm',
+    key: 'cdd6a8217c16a10e3aa2105acafc44f80ecdcd8db6d3d9df9ce39a3a4daedcee',
+    iv: 'cda28f25c83c80c5505692a4920e8ae7', plain: 'Hello World!',
+    ct: '531D1122797FC2464A43545F',
+    tag: '42BE2279090F9E3A830A3DC053CC1D90', tampered: false },
+  { algo: 'aes-256-gcm',
+    key: 'cdd6a8217c16a10e3aa2105acafc44f80ecdcd8db6d3d9df9ce39a3a4daedcee',
+    iv: 'cda28f25c83c80c5505692a4920e8ae7', plain: 'Hello World!',
+    ct: '531D1122797FC2464A43545F', aad: 'ace0dd5ad2bf',
+    tag: 'C12F114B256B4B21DD066ED0DDD743B1', tampered: false },
+  { algo: 'aes-256-gcm',
+    key: 'cdd6a8217c16a10e3aa2105acafc44f80ecdcd8db6d3d9df9ce39a3a4daedcee',
+    iv: 'cda28f25c83c80c5505692a4920e8ae7', plain: 'Hello World!',
+    ct: '421D1122673FC2464A43721A', aad: 'ace0dd5ad2bf',
+    tag: 'C12F114B256B4B21DD066ED0DDD743B1', tampered: true },
+  // 18 Byte IV
+  { algo: 'aes-128-gcm',
+    key: '6970787039613669314d623455536234',
+    iv: 'a582a11e7579bfdd0e28f008e1790990d100', plain: 'Hello World!',
+    ct: '50A054B6DA42B7A853DDE0CC',
+    tag: 'B05EEC074CE5D0984A1ECAD16300283E', tampered: false }
 ];
```

Comment:
The official GCM spec has test vectors in Appendix B.
http://csrc.nist.gov/groups/ST/toolkit/BCM/documents/proposedmodes/gcm/gcm-revised-spec.pdf
For 8bytes and 60bytes IV, there are Test Case 5/6 of aes128-gcm, Test Case 11/12 of aes192-gcm and Test Case 17/18 of aes256-gcm.
 I think it is better to use them. Also I'd be glad if you replace existing 12bytes IV tests to the official test vectors.